### ISSUE TITLE
sailfish installation

### ISSUE THREAD

**Issue by tbellembois** [therecipe/qt#893]
Hello,
I have a problem building a Sailfish app with the regular installation.
I have installed the sailfish SDK but the `qtsetup full sailfish` returns an error:

```bash
INFO[0000] running: 'qtsetup prep'                      
WARN[0000] failed to create qtrcc symlink in your PATH (/sbin/qtrcc); please use /home/thbellem/workspace/workspace_go/bin/qtrcc instead 
WARN[0000] failed to create qtmoc symlink in your PATH (/sbin/qtmoc); please use /home/thbellem/workspace/workspace_go/bin/qtmoc instead 
WARN[0000] failed to create qtminimal symlink in your PATH (/sbin/qtminimal); please use /home/thbellem/workspace/workspace_go/bin/qtminimal instead 
WARN[0000] failed to create qtdeploy symlink in your PATH (/sbin/qtdeploy); please use /home/thbellem/workspace/workspace_go/bin/qtdeploy instead 
INFO[0000] running: 'qtsetup check sailfish' [docker=false] [vagrant=false] 
INFO[0000] GOOS:                        'linux'         
INFO[0000] GOARCH:                      'amd64'         
INFO[0000] GOVERSION:                   'go1.12.6'      
INFO[0000] GOROOT:                      '/usr/local/go' 
INFO[0000] GOPATH:                   *  '/home/thbellem/workspace/workspace_go' 
INFO[0000] GOBIN:                       '/home/thbellem/workspace/workspace_go/bin' 
INFO[0000] GOMOD:                       ''              
INFO[0000] QT_HASH:                     'be44906910f46e7d43b928c84a099ad433d103c7' 
INFO[0000] QT_API:                      ''              
INFO[0000] QT_VERSION:               *  '5.12.3'        
INFO[0000] QT_DIR:                   *  '/home/thbellem/qt' 
INFO[0000] QT_STUB:                     'false'         
INFO[0000] QT_DEBUG:                    'false'         
INFO[0000] QT_QMAKE_DIR:                ''              
INFO[0000] QT_WEBKIT:                   'false'         
INFO[0000] VIRTUALBOX_DIR:              '/usr/bin'      
INFO[0000] SAILFISH_DIR:                '/home/thbellem/SailfishOS' 
INFO[0000] running: 'qtsetup generate sailfish' [docker=false] [vagrant=false] 
...

```INFO[0005] testing sailfish/listview                    
WORK=/tmp/go-build497219217
runtime/cgo
mkdir -p $WORK/b033/
cd /media/sf_GOROOT/src/runtime/cgo
CGO_LDFLAGS='"--sysroot=/srv/mer/targets/SailfishOS-2.2.1.18-armv7hl/" "-lpthread"' /media/sf_GOROOT/pkg/tool/linux_386/cgo -objdir $WORK/b033/ -importpath runtime/cgo -import_runtime_cgo=false -import_syscall=false -- -I $WORK/b033/ -g -O2 -Wall -Werror ./cgo.go
# runtime/cgo
fork/exec /srv/mer/toolings/SailfishOS-2.2.1.18/opt/cross/bin/armv7hl-meego-linux-gnueabi-gcc: no such file or directory

PANI[0024] failed to build for sailfish on linux        
panic: (*logrus.Entry) (0x966560,0xc0023c2850)

goroutine 1 [running]:
github.com/sirupsen/logrus.Entry.log(0xc0000a8240, 0xc0140af560, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/thbellem/workspace/workspace_go/src/github.com/sirupsen/logrus/entry.go:239 +0x2ce
github.com/sirupsen/logrus.(*Entry).Log(0xc0023c2000, 0x0, 0xc000475ad0, 0x1, 0x1)
        /home/thbellem/workspace/workspace_go/src/github.com/sirupsen/logrus/entry.go:268 +0xed
github.com/sirupsen/logrus.(*Entry).Logf(0xc0023c2000, 0xc000000000, 0x99877b, 0x1c, 0xc000475db0, 0x2, 0x2)
        /home/thbellem/workspace/workspace_go/src/github.com/sirupsen/logrus/entry.go:314 +0xe4
github.com/sirupsen/logrus.(*Logger).Logf(0xc0000a8240, 0x0, 0x99877b, 0x1c, 0xc000475db0, 0x2, 0x2)
        /home/thbellem/workspace/workspace_go/src/github.com/sirupsen/logrus/logger.go:145 +0x96
github.com/sirupsen/logrus.(*Logger).Panicf(...)
        /home/thbellem/workspace/workspace_go/src/github.com/sirupsen/logrus/logger.go:186
github.com/therecipe/qt/internal/cmd/deploy.build_sailfish(0x7ffe982b7e74, 0x8, 0xc001effea0, 0x65, 0x0, 0x0, 0xc001effefd, 0x8)
        /home/thbellem/workspace/workspace_go/src/github.com/therecipe/qt/internal/cmd/deploy/build.go:202 +0x1b2d
github.com/therecipe/qt/internal/cmd/deploy.build(0x96dcd8, 0x4, 0x7ffe982b7e74, 0x8, 0xc001effea0, 0x65, 0x0, 0x0, 0x0, 0x0, ...)
        /home/thbellem/workspace/workspace_go/src/github.com/therecipe/qt/internal/cmd/deploy/build.go:40 +0x2b81
github.com/therecipe/qt/internal/cmd/deploy.Deploy(0x96dcd8, 0x4, 0x7ffe982b7e74, 0x8, 0xc001effea0, 0x65, 0x0, 0x0, 0x0, 0x0, ...)
        /home/thbellem/workspace/workspace_go/src/github.com/therecipe/qt/internal/cmd/deploy/deploy.go:96 +0x1571
github.com/therecipe/qt/internal/cmd/setup.Test(0x7ffe982b7e74, 0x8, 0x0, 0x0, 0x0)
        /home/thbellem/workspace/workspace_go/src/github.com/therecipe/qt/internal/cmd/setup/test.go:160 +0x95e
main.main()
        /home/thbellem/workspace/workspace_go/src/github.com/therecipe/qt/cmd/qtsetup/main.go:120 +0x55d
```

Any idea ?

Regards,

Thomas

**Comment by therecipe** [comment]
Hey

I looked into this and this happens because the binding tries to use some old sdk version (`2.2.1.18`), instead of the recent one (`3.0.3.9`).

You usually should be able to fix this by simply using `QT_SAILFISH_VERSION=3.0.3.9`, but since I haven't looked into this quite a while there sneaked in some few small issues I had to fix with https://github.com/therecipe/qt/commit/0aa6d6dc1a6a4b1df2e7c1713d27f1192489cd6d first

To get this working you will need to update the binding and install it with `go get -v -tags=sailfish github.com/therecipe/qt/cmd/...`
You will also need to install some older Qt version first <= 5.7 (from https://download.qt.io/archive/qt/), because there were some breaking changes in the moc tool generated code in 5.8 which won't work with the Qt 5.6 libs from the SailfishOS.

(It might be possible to use newer Qt versions together with `QT_API=5.6.3` being set, if you don't make use of qtmoc, but I haven't really tested that)
The easiest way is just to install 5.6.3 or something and set `QT_VERSION=5.6.3`

Also try to shudown the sailfish build engine and the emulator before running `qtsetup full sailfish && qtsetup full sailfish-emulator`, because the binding will try to add some shared some folders to the vm, and that will only work when the VMs are stopped.

And another thing, if you update the binding without cleaning/removing the repo, then you will need to remove all the rcc* and moc* files present in `internal/examples/sailfish/listview*` and `internal/examples/quick/sailfish` because these leftovers files will break the compilation as well.

Btw, there is also a docker image which is usually easier to use
